RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -3548,6 +3548,10 @@ boolean buyUpTo(int num, item it)
 
 boolean buyUpTo(int num, item it, int maxprice)
 {
+	if(item_amount(it) >= num)
+	{
+		return true;	//we already have the target amount
+	}
```

Comment:
we were doing this check at the end to see if we reached the target amount. but that means we already returned false for "item is unbuyable"
this caused an inconsistent return value. added this initial check for consistency to always return true if we already have the desired amount